libr/io/p/io_gdb.c

```diff
@@ -102,8 +106,18 @@ static RIODesc *__open(RIO *io, const char *file, int rw, int mode) {
 	riog = R_NEW0 (RIOGdb);
 	gdbr_init (&riog->desc, false);
 	int i_port = atoi(port);
+	int i_pid = -1;
+	if (pid)
+		i_pid = atoi(pid);
+
 	if (gdbr_connect (&riog->desc, host, i_port) == 0) {
 		desc = &riog->desc;
+		desc->pid = i_pid;
+		if (pid) { // FIXME this is here for now because RDebug's pid and libgdbr's aren't properly synced.
+			int ret = gdbr_attach (desc, i_pid);
+			if (ret < 0)
```

Comment:
If you mean wrong return value because it can be both a handled error (no extended mode support) and an unhandled one (transmission error, whatever), I can fix that by return 1 instead of -2, handling 1 differently than than the hard failure.